### PR TITLE
FIX: Update Airflow Airbyte Provider to use the new Airbyte API

### LINKED ISSUE #35011: Update Airflow Airbyte Provider to use the new Airbyte API
### Description

As Airbyte transitions from the configuration API, set to sunset by early 2024, to the new Airbyte API (https://reference.airbyte.com/reference/start), it is imperative to update the existing Airflow Airbyte provider (https://airflow.apache.org/docs/apache-airflow-providers-airbyte/stable/index.html) with the new API.

@marcosmarxm maybe you have any update about this? 

### Use case/motivation

Due to anticipated changes in Airbyte API by early 2024, it's expected that the current Airflow Airbyte provider may become incompatible. Therefore, an update to the provider is necessary to ensure continued functionality

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by mitcheccles**
closes: #35011

Updated the Airbyte provider to use the new [Airbyte API](https://reference.airbyte.com/reference/start). This update ensures continued functionality as Airbyte intends to discontinue the Configuration API, which this provider perviously used to interact with Airbyte.

The discussion in issue #35011 mentioned two possible considerations when implementing a fix:

1. Use Airbyte's official [Python SDK](https://github.com/airbytehq/airbyte-api-python-sdk) 
2. Continue using the Configuration API too, for those running Airbyte v0.4x releases.

**[Airbyte's API Python SDK]**: The changes in this PR do not use the official Python SDK. This SDK is still in beta. Furthermore, I wanted to keep the changes lightweight and not add another dependency, especially considering the Airbyte Airflow provider only uses 4 of Airbyte's API endpoints.

**[Dual API Support]**:The change removes support for the Configuration API. It would have been cumbersome to support both API's. Those who are running an older version of Airbyte can install an older version of this provider when setting up Airflow.

Also worth noting, I have tested the changes in this PR on my production Airflow deployment to verify the code works.
 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/airbyte/__init__.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 __all__ = ["__version__"]
 
-__version__ = "3.6.0"
+__version__ = "3.6.1"
```
Comment (Taragolis) [inline airflow/providers/airbyte/__init__.py]: You shouldn't touch this version, it is bumped by Release Manager

**Comment by eladkal** [comment]
> The changes in this PR do not use the official Python SDK. This SDK is still in beta.

Is there official statement of Airbyte saying that? I couldn't find such statement in the SDK GitHub nor in PyPi

**Review by Taragolis — CHANGES_REQUESTED** [review]
This PR introduce breaking changes

> [!NOTE]  
> Existed users might use pretty old OSS (last year release) version of Airbyte which only support `Configuration API`

**Option 1**
If we would like to introduce  Breaking Changes then we need to create notes into the Change log and potentially simple migration guide, e.g. minimal version of Airbyte which support Official API

**Option 2**
In the other way round if we would like to keep both support for a while we need to create separate Hooks, Operators and etc (simple copy-paste) which communicate with Airbyte API only. In addition we might add UserWarning / FutureWarning into the old Hook, which raise a warning that end users use old API, which discontinued soon.

According to the recent changes into the Airbyte provider: https://github.com/apache/airflow/pull/36780 the users still use Configuration API so if I were you I would choose Option 2

File: airflow/providers/airbyte/__init__.py (hunk 1 above)
Comment (mitcheccles) [reply airflow/providers/airbyte/__init__.py]: Thanks - I've sorted that.

**Comment by mitcheccles** [comment]
> > The changes in this PR do not use the official Python SDK. This SDK is still in beta.
> 
> Is there official statement of Airbyte saying that? I couldn't find such statement in the SDK GitHub nor in PyPi
@eladkal 
https://github.com/airbytehq/airbyte-api-python-sdk?tab=readme-ov-file#maturity

**Comment by mitcheccles** [comment]
> **Option 1**
> If we would like to introduce Breaking Changes then we need to create notes into the Change log and potentially simple migration guide, e.g. minimal version of Airbyte which support Official API

@Taragolis - thanks for your comment. What do you think about asking people to downgrade their Airflow Airbyte provider in the case that someone couldn't upgrade their Airbyte deployment? - if we went for option 1 that is. It'd keep this code base clean; without copy/pasted code.

I'm happy if this PR is left open for a while - perhaps get a few more opinions.

**Comment by Taragolis** [comment]
Sorry for delay response. 

>What do you think about asking people to downgrade their Airflow Airbyte

In general it is an option, however always unknown how many users read changelogs and not blindly install latest version of provider, or just use version bundled into the Official Apache Airflow Docker image

The problem here that currently provider supports only Configurator API and that is mean that users of this provider only use it as result it might affect almost 100% current users. I'm not sure how many users would be affected, because we do not have any statistic rather than package how often package downloaded: [PyPI Stats](https://pypistats.org/packages/apache-airflow-providers-airbyte), [PePy](https://www.pepy.tech/projects/apache-airflow-providers-airbyte?versions=3.6.0&versions=3.*&versions=2.*&versions=2.*)

We always tried to deprecate something first, so end users have a chance to smooth transition to the new API.

In addition we have no idea how many users extend AirbyteHook as well as related operators, so it might take a time to them to migrate.

**Comment by potiuk** [comment]
If it was a simple breaking change in parameters that users could correct in their DAGs. I'd be for immediate breaking change. But the fact that users might be using an older version of airbyte OSS makes a big difference, because that might require way more effort to upgrade.

As I see it - this change is not breaking for DAG Authoring but breaking for - potentially - deployment. 

Do we know how old version of the OSS Airbyte does not support the new API ? How old it is? Any stats of usage that Airbyte might have?

I think we should assess it based on likelihood of breaking multiple users workflows.  If the OSS version of Airbyte that is incompatible is like 2 years old, for example, the likelihood is small. Then we could capture the error that happens in this case and explicitly ask users to downgrade. To @Taragolis comment - as long as we raise a very explicitt error and clearly communicate the user "Your Airbyte OSS version does not support the new API this provider supports, please downgrade porovider to version x.y.z" - or something like that - then option 1 is acceptable. But if someone's OSS fresh version installed 4 months ago will stop working then option 2 is better.

**Comment by Taragolis** [comment]
Just for the record, I've checked latest Airbyte `0.50.48` locally and both APIs available. I'm not sure but last time I've found that presumably [v0.50.13 ](https://github.com/airbytehq/airbyte-platform/releases/tag/v0.50.13) (released 5 month ago) it is the first version where Airbyte API available for OSS users 

**Configuration API**
```console
❯ curl -X POST "http://127.0.0.1:8000/api/v1/workspaces/list" \
     --header 'accept: application/json' \
     --user airbyte:password

{"workspaces":[{"workspaceId":"c81b0f84-e935-4ca4-974f-b1dbc8e4af58","customerId":"3aa7d861-9af9-4708-86ea-cd089605ef6e","email":"example@org.org","name":"c81b0f84-e935-4ca4-974f-b1dbc8e4af58","slug":"c81b0f84-e935-4ca4-974f-b1dbc8e4af58","initialSetupComplete":true,"displaySetupWizard":false,"anonymousDataCollection":false,"notifications":[],"notificationSettings":{},"defaultGeography":"auto","organizationId":"00000000-0000-0000-0000-000000000000"}]}
```
**Airbyte API**
```console
❯ curl -X GET "http://127.0.0.1:8006/v1/workspaces?includeDeleted=false&limit=20&offset=0" \
     --header 'accept: application/json' \
     --user airbyte:password

{"data":[{"workspaceId":"c81b0f84-e935-4ca4-974f-b1dbc8e4af58","name":"c81b0f84-e935-4ca4-974f-b1dbc8e4af58","dataResidency":"auto"}]}
```

**Comment by potiuk** [comment]
so it's ~ the threshold I had in mind. I'd say it's a bit too early to remove it then and supporting both should be preferred

**Pull request closed without merging**